cognite/neat/rules/models/rules.py

```diff
@@ -657,24 +663,27 @@ def is_raw_lookup(self) -> bool:
     def replace_float_nan_with_default(cls, values: dict) -> dict:
         return replace_nan_floats_with_default(values, cls.model_fields)
 
-    @validator("expected_value_type", pre=True)
-    def replace_string_with_entity(cls, value):
-        # handles simple types
+    @field_validator("container", mode="before")
+    def container_string_to_entity(cls, value):
+        if value:
+            try:
+                return Container.from_string(entity_string=value)
+            except ValueError:
+                return Container(prefix="undefined", suffix=value, name=value)
```

Comment:
My personal preference is to avoid indent as much as possible
```suggestion
        if not value:
            return value
        try:
            return Container.from_string(entity_string=value)
        except ValueError:
            return Container(prefix="undefined", suffix=value, name=value
```